xds/internal/xdsclient/filter_chain_test.go

```diff
@@ -377,6 +417,352 @@ func TestNewFilterChainImpl_Failure_BadSecurityConfig(t *testing.T) {
 	}
 }
 
+// TestNewFilterChainImpl_Failure_BadHTTPFilters verifies cases where the HTTP
+// Filters in the filter chain are invalid.
+func TestNewFilterChainImpl_Failure_BadHTTPFilters(t *testing.T) {
+	tests := []struct {
+		name    string
+		lis     *v3listenerpb.Listener
+		wantErr string
+	}{
+		{
+			name: "client side HTTP filter",
+			lis: &v3listenerpb.Listener{
+				Name:    "grpc/server?xds.resource.listening_address=0.0.0.0:9999",
+				Address: &v3corepb.Address{Address: &v3corepb.Address_SocketAddress{SocketAddress: &v3corepb.SocketAddress{Address: "0.0.0.0", PortSpecifier: &v3corepb.SocketAddress_PortValue{PortValue: 9999}}}},
```

Comment:
You probably don't need the address field in these tests since the address field checks are made in `processServerSideListener` before `NewFilterChainManager` is called.

Reply:
Ah, good catch, thanks! Deleted.